app/views/play/level/tome/SpellView.coffee

```diff
@@ -106,6 +107,21 @@ module.exports = class SpellView extends CocoView
     $(@ace.container).find('.ace_gutter').on 'click', @onGutterClick
     @initAutocomplete aceConfig.liveCompletion ? true
 
+    # Create a Spade to 'dig' into Ace.
+    if Spade
+      spade = @spade = new Spade()
+      @spade.track(@ace)
+
+      # If a user is taking longer than 10 minutes, let's log it.
+      tenMin = 10 * 60 * 1000
+      view = @
+      @spadeWatch = setTimeout(->
```

Comment:
Better style and garbage collection like this:

```
    spadeSaveDelay = 10 * 60 * 1000
    @spadeSaveTimeout = setTimeout @putSpade, spadeSaveDelay

  saveSpade: =>
    return if @destroyed
```

Call it `saveSpade` instead of `putSpade`? Also in the view's `destroy` method, clear the `@spadeSaveTimeout`.